### PR TITLE
Fix migrate script for MacOS

### PR DESCRIPTION
**Pull request by Kshitij09**
## Purpose / Description
Fixed migrate script for the MacOS and tried migrating FieldEditLine with it. (_Will raise a new PR following Comment draft and only commit fixing the script_)

## Approach
Will brief about it tomorrow

## How Has This Been Tested?
By building the project on M1 Mac

## Learning (optional, can help others)
* https://unix.stackexchange.com/a/205854
* https://stackoverflow.com/a/4247333

### CODE REVIEW

**Comment by Kshitij09** [comment]
@Arthur-Milchior Please review!

**Comment by Kshitij09** [comment]
[Migrate_FieldEditLine.patch](https://drive.google.com/file/d/146mvMgGPnf72HXzVSoLDTVJ5tP41vsk9/view?usp=sharing) Use this patch to try out the script.
### Steps to follow
1. Drop the last 2 commits
2. `git apply Migrate_FieldEditLine.patch`
3. `git rm AnkiDroid/src/main/java/com/ichi2/anki/FieldEditLine.java`
4. `git add AnkiDroid/src/main/java/com/ichi2/anki/FieldEditLine.kt`
5. `./tools/migrate.sh`

File: tools/migrate.sh [hunk 1]
```diff
@@ -13,29 +13,27 @@
...
-#################
-
-if ["$OSTYPE" =~ ^darwin];
-then
-    echo "The script do not work on Mac OS."
-    echo "If you know how to correct it, it would be greatly appreciated."
-    echo "If you see this error and are not on mac, something went wrong."
-    exit 1
-fi;
+function sedcompat() {
```
Comment (Arthur-Milchior) [inline tools/migrate.sh:16]: Why do you define this function and never use it?

**Review by Arthur-Milchior — COMMENTED** [review]
Thanks a lot. I tested and it still works on my linux too, which is a really great news.

I'd love to know why those changes were useful.

In particular, I wanted to add 

`sed -i "s~<\([^>]*\)></\1*>~<\1~" $FILEPATH_KT`
to ensure we remove the duplicate email start and <http:>, but I can't do it now as I don't know how it will impact mac

**Comment by Arthur-Milchior** [comment]
on linux, you need to use `sed -i.bak`, without spaces. Otherwise it believes `.bak` is the sed instruction and sed is not applied

File: tools/migrate.sh (hunk 1 above)
Comment (Kshitij09) [reply tools/migrate.sh:16]: Oh, it was a failed attempt, forgot to remove it.

**Pull request closed without merging**